controllers/activegate/capability/capability.go

```diff
@@ -146,10 +190,14 @@ func NewRoutingCapability(crProperties *dynatracev1alpha1.CapabilityProperties)
 			},
 		},
 	}
+
+	setTlsConfig(agSpec, &c.capabilityBase)
```

Comment:
please add unit tests to check whether tls config is applied correctly if a secret name is provided and that it is not applied if no secret name is given.

Reply:
Added UT for first scenario.
Regarding second scenario - just by definition, secret cannot be applied if no secret name is given.

Reply:
my use case for the second scenario is the following:

In the future, some developer is working on the setTlsConfig.
They change some stuff here, maybe refactor something there.
Then they forget to reinstate the  check `if agSpec.TlsSecretName != "" {`
Now a VolumeSource is always created, even with an empty SecretName, breaking the deployment of the capability

In my mind, that is a mistake that is easily catch-able by a simple unit test.

Reply:
Ok, I get it now. It's already covered in `"default"` [case](https://github.com/Dynatrace/dynatrace-operator/pull/240/files#diff-ab178ad5f17da576726643ba0e3f6e9c442efc4f2dc621d2d4c40267f2108f29R256), where `want` has no `VolumeMount` set (`nil`).
Test [fails](https://github.com/Dynatrace/dynatrace-operator/pull/240/files#diff-ab178ad5f17da576726643ba0e3f6e9c442efc4f2dc621d2d4c40267f2108f29R313) if `VolumeMount` is created because `reflect.DeepEqual` returns false as `VolumeMount` was not expected.